### PR TITLE
PostgreSQL And OpenGauss Support Get Rename Column

### PR DESCRIPTION
**Pull request by ediconss**
Fixes https://github.com/apache/shardingsphere/issues/27957

Changes proposed in this pull request:
  -

---

Before committing this PR, I'm sure that I have checked the following options:
- [ ] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [ ] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [ ] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added corresponding unit tests for my changes.

### CODE REVIEW

**Comment by strongduanmu** [comment]
Hi @ediconss, can you refer the original issue link?

File: parser/sql/dialect/postgresql/src/main/java/org/apache/shardingsphere/sql/parser/postgresql/visitor/statement/type/PostgreSQLDDLStatementVisitor.java [hunk 1]
```diff
@@ -419,6 +419,9 @@ public ASTNode visitAlterDefinitionClause(final AlterDefinitionClauseContext ctx
         if (null != ctx.alterTableActions()) {
             result.getValue().addAll(ctx.alterTableActions().alterTableAction().stream().flatMap(each -> getAlterDefinitionSegments(each).stream()).collect(Collectors.toList()));
         }
+        if (null != ctx.renameColumnSpecification()) {
```
Comment (strongduanmu) [inline parser/sql/dialect/postgresql/src/main/java/org/apache/shardingsphere/sql/parser/postgresql/visitor/statement/type/PostgreSQLDDLStatementVisitor.java:422]: Can you add sql parser test for this change?

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: parser/sql/dialect/postgresql/src/main/java/org/apache/shardingsphere/sql/parser/postgresql/visitor/statement/type/PostgreSQLDDLStatementVisitor.java (hunk 1 above)
Comment (strongduanmu) [inline parser/sql/dialect/postgresql/src/main/java/org/apache/shardingsphere/sql/parser/postgresql/visitor/statement/type/PostgreSQLDDLStatementVisitor.java:422]: Please add sql parser test case for this change @ediconss , thank you.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: parser/sql/dialect/postgresql/src/main/java/org/apache/shardingsphere/sql/parser/postgresql/visitor/statement/type/PostgreSQLDDLStatementVisitor.java (hunk 1 above)
Comment (ediconss) [reply parser/sql/dialect/postgresql/src/main/java/org/apache/shardingsphere/sql/parser/postgresql/visitor/statement/type/PostgreSQLDDLStatementVisitor.java:422]: sorry,I don't know where to add，parser/sql/dialect/postgresql/src/test/java/org/apache/shardingsphere/test/it/sql/parser/it/postgresql/external/ExternalPostgreSQLParserIT.java already covered  relname col sql

Comment (strongduanmu) [reply parser/sql/dialect/postgresql/src/main/java/org/apache/shardingsphere/sql/parser/postgresql/visitor/statement/type/PostgreSQLDDLStatementVisitor.java:422]: You can refer InternalSQLParserIT, there are lots of sql parse case now.

Comment (ediconss) [reply parser/sql/dialect/postgresql/src/main/java/org/apache/shardingsphere/sql/parser/postgresql/visitor/statement/type/PostgreSQLDDLStatementVisitor.java:422]: https://github.com/apache/shardingsphere/blob/0a62ba2670f3eee6becf1e2aebaf1817c60282ee/test/it/parser/src/main/resources/sql/supported/ddl/alter-table.xml#L112

Comment (strongduanmu) [reply parser/sql/dialect/postgresql/src/main/java/org/apache/shardingsphere/sql/parser/postgresql/visitor/statement/type/PostgreSQLDDLStatementVisitor.java:422]: So why this test case not find this bug? I think we should assert the new visit segment.

Comment (ediconss) [reply parser/sql/dialect/postgresql/src/main/java/org/apache/shardingsphere/sql/parser/postgresql/visitor/statement/type/PostgreSQLDDLStatementVisitor.java:422]: https://github.com/apache/shardingsphere/blob/0a62ba2670f3eee6becf1e2aebaf1817c60282ee/test/it/parser/src/main/resources/case/ddl/alter-table.xml#L972-L974i , no assert

**Review by strongduanmu — APPROVED** [review]

**Pull request merged**